### PR TITLE
DM21 Architecture

### PR DESCRIPTION
**Pull request by GreatRSingh**
## Description

Fix #(issue)

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by aaronrockmenezes — CHANGES_REQUESTED** [review]

File: deepchem/models/dft/dm21.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        # Layer Initialisation
+        self.lin_tanh = nn.Linear(11, self.hidden_size)
+        self.lin_elu = nn.ModuleList()
+        for i in range(self.n_layers):
+            self.lin_elu.append(nn.Linear(self.hidden_size, self.hidden_size))
+        self.final = nn.Linear(self.hidden_size, 3)
+        self.acti_tanh = nn.Tanh()
+        self.acti_elu = nn.ELU()
+        self.acti_scaled_sigmoid = nn.Sigmoid()
```
[the remark this answers is not in the document]
Comment (GreatRSingh) [reply deepchem/models/dft/dm21.py:82]: Thanks for pointing it out, I have made some changes can you verify if its correct.

File: deepchem/models/torch_models/layers.py [hunk 2]
```diff
@@ -6477,3 +6477,51 @@ def forward(self, inputs: List[np.ndarray]):
         if self.activation_fn is not None:
             mol_features = self.activation_fn(mol_features)
         return mol_features
+
+
+class Logistic(nn.Module):
+    """A logistic function or logistic curve is a common S-shaped curve (sigmoid curve).
+    Logistic functions are often used in artificial neural networks to introduce
+    nonlinearity in the model or to clamp signals to within a specified interval.
```
Comment (aaronrockmenezes) [inline deepchem/models/torch_models/layers.py:6485]: Can you add an example showing how a user can have a logistic layer output in a different interval than the normal (0,1)?

File: deepchem/models/dft/dm21.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aaronrockmenezes) [reply deepchem/models/dft/dm21.py]: I can't see the change. Could you point out where it is? I think the nn.Sigmoid function needs to be changed.

File: deepchem/models/torch_models/layers.py (hunk 2 above)
Comment (GreatRSingh) [reply deepchem/models/torch_models/layers.py:6485]: 😀 thanks for suggesting, will add it.
